doc/modules/ensemble.rst

```diff
@@ -1,534 +1,437 @@
+to split a node into child nodes. This creates so called interactions between
+features, i.e. usage of different features as split along a branch. Sometimes,
+one wants to restrict the possible interactions, see [Mayer2022]_. This can be
+done by the parameter ``interaction_cst``, where one can specify the indices
+of features that are allowed to interact.
+For instance, with 3 features in total, ``interaction_cst=[{0}, {1}, {2}]``
+forbids all interactions.
+The constraints ``[{0, 1}, {1, 2}]`` specifies two groups of possibly
+interacting features. Features 0 and 1 may interact with each other, as well
+as features 1 and 2. But note that features 0 and 2 are forbidden to interact.
+The following depicts a tree and the possible splits of the tree:
 
-Totally Random Trees Embedding
-------------------------------
+.. code-block:: none
 
-:class:`RandomTreesEmbedding` implements an unsupervised transformation of the
-data.  Using a forest of completely random trees, :class:`RandomTreesEmbedding`
-encodes the data by the indices of the leaves a data point ends up in.  This
-index is then encoded in a one-of-K manner, leading to a high dimensional,
-sparse binary coding.
-This coding can be computed very efficiently and can then be used as a basis
-for other learning tasks.
-The size and sparsity of the code can be influenced by choosing the number of
-trees and the maximum depth per tree. For each tree in the ensemble, the coding
-contains one entry of one. The size of the coding is at most ``n_estimators * 2
-** max_depth``, the maximum number of leaves in the forest.
+      1      <- Both constraint groups could be applied from now on
+     / \
+    1   2    <- Left split still fulfills both constraint groups.
+   / \ / \      Right split at feature 2 has only group {1, 2} from now on.
 
-As neighboring data points are more likely to lie within the same leaf of a
-tree, the transformation performs an implicit, non-parametric density
-estimation.
+LightGBM uses the same logic for overlapping groups.
 
-.. topic:: Examples:
+Note that features not listed in ``interaction_cst`` are automatically
+assigned an interaction group for themselves. With again 3 features, this
+means that ``[{0}]`` is equivalent to ``[{0}, {1, 2}]``.
 
- * :ref:`sphx_glr_auto_examples_ensemble_plot_random_forest_embedding.py`
+.. topic:: References
 
- * :ref:`sphx_glr_auto_examples_manifold_plot_lle_digits.py` compares non-linear
-   dimensionality reduction techniques on handwritten digits.
+  .. [Mayer2022] M. Mayer, S.C. Bourassa, M. Hoesli, and D.F. Scognamiglio.
+     2022. :doi:`Machine Learning Applications to Land and Structure Valuation
+     <10.3390/jrfm15050193>`.
+     Journal of Risk and Financial Management 15, no. 5: 193
 
- * :ref:`sphx_glr_auto_examples_ensemble_plot_feature_transformation.py` compares
-   supervised and unsupervised tree based feature transformations.
+Low-level parallelism
```

Comment:
I'm not a fan of "low-level". Low level makes me think of C and Assembler. Things that are complicated to use and better left to experts.

What about "Built-in parallelism"?

Reply:
This parallelism is lower-level than the one that we typically expose. It is indeed in C.

I would rather not change this, in particular since it is not related to the changes that I propose.